src/DependencyInjection/Security/Factory/BearerFactory.php

```diff
@@ -26,25 +32,50 @@ public function create(
             ->replaceArgument(0, new Reference($userProvider));
 
         $listenerId = 'security.authentication.listener.bearer.'.$id;
-        $listener = $container->setDefinition(
+
+        $container->setDefinition(
             $listenerId,
             new ChildDefinition('paysera_bearer_authentication.listener.bearer_listener')
         );
 
-        return [$providerId, $listenerId, $defaultEntryPoint];
+        return [
+            $providerId,
+            $listenerId,
+            $defaultEntryPointId,
+        ];
     }
 
-    public function getPosition()
+    /**
+     *
+     * @deprecated since Symfony 5.4, use AuthenticatorFactoryInterface instead
+     */
+    public function getPosition(): string
     {
         return 'pre_auth';
     }
 
-    public function getKey()
+    public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): string
+    {
+        $authenticatorId = 'security.authentication.provider.bearer.' . $firewallName;
+
+        $authenticator = (new ChildDefinition('paysera_bearer_authentication.authenticator.bearer_passport'));
+
+        $container->setDefinition($authenticatorId, $authenticator);
+
+        return $authenticatorId;
+    }
+
+    public function getPriority(): int
+    {
+        return -10;
```

Comment:
Is it default value? If not, why `-10` ?

Reply:
it's not the default value. -10 means it is high priority if multiple authenticators are present. it can be `0` too

Reply:
-10 is the same as the previous priority `pre_auth`.  [https://github.com/symfony/security-bundle/blob/5.4/DependencyInjection/SecurityExtension.php#L1110]( https://github.com/symfony/security-bundle/blob/5.4/DependencyInjection/SecurityExtension.php#L1110)